apps/researcher/src/app/[locale]/communities/[id]/buttons.tsx

```diff
@@ -0,0 +1,89 @@
+'use client';
+
+import {useState} from 'react';
+import {useTranslations} from 'next-intl';
+import {useClerk, useUser} from '@clerk/nextjs';
+import {useTransition} from 'react';
+import {joinCommunity} from './actions';
+
+interface Props {
+  organization: {
+    id: string;
+  };
+}
+
+// If logged in and not part of the community, show the join button
+export function JoinCommunityButton({organization}: Props) {
+  const {isLoaded, isSignedIn, user} = useUser();
+  const [isClicked, setIsClicked] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [isPending, startTransition] = useTransition();
+
+  const userIsMember = user?.organizationMemberships.some(
+    membership => membership.organization.id === organization.id
+  );
+
+  const showJoinButton = isLoaded && isSignedIn && !userIsMember;
+
+  const joinCommunityClick = () => {
+    setIsClicked(true);
+    startTransition(async () => {
+      try {
+        await joinCommunity({
+          organizationId: organization.id,
+          userId: user!.id,
```

Comment:
Minor: this uses `!` but line 22 uses `?` - intentionally?

Reply:
Yes. I don't want an error if `user` is undefined on row 22. The button will be hidden because `isSignedIn` then is false.

I do want an error if there is no user when trying to save.

But I will change the logic of `showJoinButton` a bit so it will be false if there is no user, which is probably more precise, without making the assumption that `isSignedIn` is false if `user` is undefined.